worldedit-core/src/main/java/com/sk89q/worldedit/util/net/HttpRequest.java

```diff
@@ -387,26 +402,51 @@ private Form() {
          * @return this object
          */
         public Form add(String key, String value) {
-            try {
-                elements.add(URLEncoder.encode(key, "UTF-8")
-                    + "=" + URLEncoder.encode(value, "UTF-8"));
-                return this;
-            } catch (UnsupportedEncodingException e) {
-                throw new RuntimeException(e);
+            elements.put(key, value);
+            return this;
+        }
+
+        public String getFormDataSeparator() {
+            return formDataSeparator.getValue();
+        }
+
+        public String toFormDataString() {
+            String formSeparator = "--" + formDataSeparator.getValue();
+            StringBuilder builder = new StringBuilder();
+
+            for (Map.Entry<String, String> element : elements.entrySet()) {
+                builder
+                    .append(formSeparator)
+                    .append("\r\n")
+                    .append("Content-Disposition: form-data; name=\"")
+                    .append(element.getKey())
+                    .append("\"\r\n\r\n")
+                    .append(element.getValue())
+                    .append("\r\n");
             }
+
+            builder.append(formSeparator).append("--");
+
+            return builder.toString();
         }
 
-        @Override
-        public String toString() {
+        public String toUrlEncodedString() {
             StringBuilder builder = new StringBuilder();
             boolean first = true;
-            for (String element : elements) {
+            for (Map.Entry<String, String> element : elements.entrySet()) {
```

Comment:
This might also use Joiner, esp. MapJoiner.

Reply:
This would be fairly messy to do, due to the need of URLEncoding the values/keys, and the fact that it has a checked exception on it